### DISCUSSION TITLE
Proposal: Iterator over all subgraphs/subtrees with max weight

### DISCUSSION THREAD

**Discussion by hjtran** [networkx/networkx#5931] [Algorithms]
I've implemented an algorithm that iterates over all subtrees (and subgraphs since it doesn't take much more to add that) of a graph `G` with max weight `m` (for my use case, I also included a "core" subgraph (`C`) which must be preserved for all other subgraphs).

Trivially you could do this by iterating over all subgraphs and just checking its weight and throwing it out if it doesn't meet the correct weight but this iterating over many subgraphs that don't meet the criterion before seeing a subgraph that does. The algorithm I've implemented just builds up subgraphs that meet the criteria and prunes out trees that'd consider too large of subgraphs.

Just wanted to see if this is something that'd be of interest for inclusion with `networkx` (or maybe it's already been implemented and I missed it and you can point it out to me :-)). I get that y'all probably don't ship just any and all arbitrary algorithms so also cool if it doesn't seem widely useful enough, just wanted to check.

**Comment by juanis2112** [comment]
Hi @hjtran, thanks for wanting to contribute to NetworkX! According to our contributing guide https://github.com/networkx/networkx/blob/main/CONTRIBUTING.rst#adding-references if you want to contribute a new algorithm you should also provide a reference paper or resource. Please let us know if you can provide this. If that's not the case, maybe if you can find a relevant use case,  it would be better to add it as an example to the gallery. 

@rossbar ideas?

**Reply by rossbar** [reply]
I tend to agree with @juanis2112 - the general "policy" (such as it is) for deciding what should be included is to try to develop a sense of the usefulness/ubiquity of the operation. One obvious way to do that is by referencing published work - i.e. an established, well-known article out there that may give us a sense for how useful it would be to include a new function. That's not the *only* consideration however! In this case, the problem you've described seems like a pretty general one... perhaps it has a very common use-case? If so, sharing that would also provide motivation for inclusion.

The final thing to consider is: how complicated is the algorithm? Is it only a few lines that rely mostly on existing networkx functions? If so, it may be a better fit as a e.g. a [gallery example](https://networkx.org/documentation/latest/auto_examples/index.html) or an example in a docstring.
